eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -62,6 +62,7 @@ class Router(val nodeParams: NodeParams, watcher: typed.ActorRef[ZmqWatcher.Comm
   context.system.eventStream.subscribe(self, classOf[LocalChannelUpdate])
   context.system.eventStream.subscribe(self, classOf[LocalChannelDown])
   context.system.eventStream.subscribe(self, classOf[AvailableBalanceChanged])
+  context.system.eventStream.publish(IAmReady(this.getClass))
```

Comment:
Shouldn't we emit `IAmReady` just before `startWith`, once we've finished initializing all data?

Reply:
All messages sent to the router will wait in the actor's mailboc. I chose to keep `IAmReady` close to the `eventStream` subscriptions because this is what actually matters. Maybe I should rename `IAmReady` to `SubscriptionsComplete` so the semantics are more clear?

Reply:
> Maybe I should rename IAmReady to SubscriptionsComplete so the semantics are more clear?

Yes, I think it would make more sense.

Reply:
Done in [9ca8551](https://github.com/ACINQ/eclair/pull/2322/commits/9ca8551a1d16a2cec2acb1ea46373a3d9266946e).